library/core/tests/ptr.rs

```diff
@@ -455,6 +455,18 @@ fn align_offset_various_strides() {
     assert!(!x);
 }
 
+#[test]
+fn align_offset_issue_103361() {
+    #[cfg(target_pointer_width = "64")]
+    const SIZE: usize = 1 << 47;
+    #[cfg(target_pointer_width = "32")]
+    const SIZE: usize = 1 << 30;
+    #[cfg(target_pointer_width = "16")]
+    const SIZE: usize = 1 << 13;
+    struct HugeSize([u8; SIZE - 1]);
+    let _ = (SIZE as *const HugeSize).align_offset(SIZE);
```

Comment:
We usually prefer the strict provenance APIs in libcore -- https://github.com/rust-lang/rust/pull/104632

Note sure if the lint against int2ptr casts ever got implemented? If yes we should probably enable it here.

Reply:
Ah, I basically just copy-pasted over the reproducer from the issue…